### ISSUE TITLE
Massive build errors in ATDM 'cee-rhel6' builds starting 2019-10-24

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#6040]
CC: @trilinos/<package-name>, @<product-area-lead> (Trilinos <product-area-name> Product Lead), @bartlettroscoe, @fryeguy52

<Checklist>
<???: Add label "client: ATDM">
<???: Add label "ATDM Sev: Blocker" (by default but could be other "ATDM Sev: XXX")>
<???: Add label "type: bug"?>
<???: Add label for affected packages (e.g. "pkg: MueLu", "pkg: Tpetra", "pkg: Kokkos", etc.)>
<???: Add label "PA: ???Project Area???" (e.g. "PA: Linear Solvers", "PA: Data Services")>
<???: Add milestone "Initial cleanup of new ATDM ..." or "Keep promoted ATDM ...">
<???: Once GitHub Issue is created, add entries for tests to TrilinosATDMStatus/*.csv files>

## Next Action Status

<status-and-or-first-action>

## Description

As shown in [this query](https://testing.sandia.gov/cdash/index.php?project=Trilinos&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-cee-rhel6&field2=buildname&compare2=65&value2=Trilinos-atdm-) there are mass build errors in all of the ATDM Trilinos 'cee-rhel6' builds:

* Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt
* Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt
* Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt
* Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt

Only the install failures show up on CDash.  For some reason, the actually build errors themselves are not showing up on CDAsh.  (I know this because the build times for these builds are between 9s and 13s which mans that only the 'install_package_by_package' target build results are showing up).

## Current Status on CDash

???

## Steps to Reproduce

I will put in later once I know what packages these are (as I can't see build errors on CDash).

**Comment by bartlettroscoe** [comment]
@jfike, @sebrowne, @micahahoward,

FYI: Looks like the upgrade of SuperLUDist from 4.2 to 5.4.0 broke these ('lib' dir changed to 'lib64').  This will result in all of the SPARC Trilinos Integration builds failing tomorrow that point to these installs for Trilinos testing day 2019-10-04 (that don't exist).

I think I have this fixed in my local branch and I will post PR once I get confirmation.

P.S. Looks like there is a defect in CDash in that it does not show build errors due to missing dependencies:

```
ninja: error: '/projects/sparc/tpls/cee-rhel6/superlu_dist-5.4.0/a3121eaff44f7bf7d44e625c3b3d2a9911e58876/cee-cpu_clang-5.0.1_openmpi-1.10.2/lib/libsuperlu_dist.a', needed by 'packages/belos/test/BlockCG/Belos_bl_cg_real_hb.exe', missing and no known rule to make it
```

**Comment by bartlettroscoe** [comment]
@jfike, @sebrowne, @micahahoward,

This is fixed in PR #6048.  I just manually merged to 'atdm-nightly' in commit 
debdf03 so this will get run in the ATDM Trilinos builds tomorrow.

The ATDM Trilinos Integration builds should be fixed on Sunday (SPARC testing day 2019-10-06).

<details>
<summary><b>DETAILED NOTES</b> (click to expand)</summary>

.

Trying to reproduce build errors with just Kokkos on 'ceerws1113' with Trilinos versin

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CHECKIN/

$ ./checkin-test-atdm-cee-rhel6.sh \
  cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt \
  --enable-packages=Kokkos --local-do-all

...

PASSED (NOT READY TO PUSH): Trilinos: ceerws1113

Fri Oct  4 10:29:12 MDT 2019

Enabled Packages: Kokkos

Build test results:
-------------------
0) MPI_RELEASE_DEBUG_SHARED_PT_OPENMP => Test case MPI_RELEASE_DEBUG_SHARED_PT_OPENMP was not run! => Does not affect push readiness! (-1.00 min)
1) cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt => passed: passed=24,notpassed=0 (2.04 min)
```

I could not reproudce any failures, at least not in Kokkos.

I am wondering if this might be an env issue.

I will try running a local build and install myself with the driver on 'ceerws1113' with:

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/

$ env \
    ATDM_TRILINOS_INSTALL_PREFIX=install \
    CTEST_DO_INSTALL=ON \
    ATDM_CTEST_S_USE_FULL_BUILD_NAME=1 \
  ./ctest-s-local-test-driver-cee-rhel6.sh \
    Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt


***
*** /scratch/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/scratch/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'ceerws1113' matches known ATDM host 'cee-rhel6' and system 'cee-rhel6'
Setting compiler and build options for build-name 'cee-rhel6-default'
Using CEE RHEL6 compiler stack CLANG-5.0.1_OPENMPI-1.10.2 to build DEBUG code with Kokkos node type SERIAL

Running builds:
    Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt

Running Jenkins driver Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt/smart-jenkins-driver.out

real    2m8.887s
user    1m25.134s
sys     0m38.063s
```

Shoot, that failed with a build error (that got written over when the 'install' occurred).

I will run again without the install this time:

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/

$ env \
    ATDM_TRILINOS_INSTALL_PREFIX=install \
    ATDM_CTEST_S_USE_FULL_BUILD_NAME=1 \
  ./ctest-s-local-test-driver-cee-rhel6.sh \
    Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt

real    2m2.087s
user    1m24.171s
sys     0m39.920s
```

Okay, now I can see the error:

```
ninja: error: '/projects/sparc/tpls/cee-rhel6/superlu_dist-5.4.0/a3121eaff44f7bf7d44e625c3b3d2a9911e58876/cee-cpu_clang-5.0.1_openmpi-1.10.2/lib/libsuperlu_dist.a', needed by 'packages/belos/test/BlockCG/Belos_bl_cg_real_hb.exe', missing and no known rule to make it
```

in the file:

```
/scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt/SRC_AND_BUILD/BUILD/Testing/Temporary/LastBuild_20191004-1647.log
```

but that did not get dispayed on CDash :-(

Looks like the dir 'lib' change to 'lib64':

```
$ ls /projects/sparc/tpls/cee-rhel6/superlu_dist-5.4.0/a3121eaff44f7bf7d44e625c3b3d2a9911e58876/cee-cpu_clang-5.0.1_openmpi-1.10.2/lib64/
libsuperlu_dist.a  pkgconfig
```

Should be an easy fix.

I can test that locally with:

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CHECKIN/

$ ./checkin-test-atdm-cee-rhel6.sh \
  cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt \
  --enable-packages=Amesos2 --local-do-all

...

PASSED (NOT READY TO PUSH): Trilinos: ceerws1113

Fri Oct  4 11:02:08 MDT 2019

Enabled Packages: Amesos2

Build test results:
-------------------
0) MPI_RELEASE_DEBUG_SHARED_PT_OPENMP => Test case MPI_RELEASE_DEBUG_SHARED_PT_OPENMP was not run! => Does not affect push readiness! (-1.00 min)
1) cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt => passed: passed=8,notpassed=0 (2.75 min)
```

That passed.

Now to run the builds again:

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/

$ env \
    ATDM_TRILINOS_INSTALL_PREFIX=install \
    CTEST_DO_INSTALL=ON \
  ./ctest-s-local-test-driver-cee-rhel6.sh \
    all

***
*** /scratch/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/scratch/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'ceerws1113' matches known ATDM host 'cee-rhel6' and system 'cee-rhel6'
Setting compiler and build options for build-name 'cee-rhel6-default'
Using CEE RHEL6 compiler stack CLANG-5.0.1_OPENMPI-1.10.2 to build DEBUG code with Kokkos node type SERIAL

Running builds:
    cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt
    cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt
    cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt
    cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt

Running Jenkins driver Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt.sh ...

    See log file Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt/smart-jenkins-driver.out


real    64m7.581s
user    763m29.650s
sys     37m4.853s

Running Jenkins driver Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt/smart-jenkins-driver.out

real    57m20.504s
user    698m24.855s
sys     57m3.307s

Running Jenkins driver Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt/smart-jenkins-driver.out

real    137m55.226s
user    1886m56.710s
sys     193m18.534s

Running Jenkins driver Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt/smart-jenkins-driver.out

real    107m45.616s
user    1543m57.330s
sys     121m46.679s
```

Darn, I set the wrong var.  The right var to set is `ATDM_CONFIG_TRIL_CMAKE_INSTALL_PREFIX`.  I did not read my own documentation very well :-(

In any case, it looks like everything is building and running just fine as shown at:

* https://testing.sandia.gov/cdash/index.php?project=Trilinos&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-cee-rhel6

It is just that the installs are failing because it is trying to install under the default CMAKE_INSTALL_PREFIX which is /user/local/.

I am going to rerun with a rebuild to prove the install works with:

```
$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/

$ env \
    CTEST_START_WITH_EMPTY_BINARY_DIRECTORY=FALSE \
    ATDM_CONFIG_TRIL_CMAKE_INSTALL_PREFIX=install \
    CTEST_DO_INSTALL=ON \
  ./ctest-s-local-test-driver-cee-rhel6.sh \
    all

***
*** /scratch/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/scratch/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'ceerws1113' matches known ATDM host 'cee-rhel6' and system 'cee-rhel6'
Setting compiler and build options for build-name 'cee-rhel6-default'
Using CEE RHEL6 compiler stack CLANG-5.0.1_OPENMPI-1.10.2 to build DEBUG code with Kokkos node type SERIAL

Running builds:
    cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt
    cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt
    cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt
    cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt

Running Jenkins driver Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt.sh ...

    See log file Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt/smart-jenkins-driver.out

real    15m43.390s
user    46m8.270s
sys     5m7.580s

Running Jenkins driver Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt.sh ...

    See log file Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt/smart-jenkins-driver.out

real    13m19.842s
user    64m45.062s
sys     15m28.727s

Running Jenkins driver Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt.sh ...

    See log file Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt/smart-jenkins-driver.out

real    10m44.411s
user    95m36.644s
sys     12m28.003s

Running Jenkins driver Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt.sh ...

    See log file Trilinos-atdm-cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt/smart-jenkins-driver.out

real    9m30.707s
user    112m18.941s
sys     12m23.992s
```

That time everything passed, including the installs as shown in the most recent builds at:

* 
https://testing.sandia.gov/cdash/index.php?project=Trilinos&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-cee-rhel6

(except for a single failing test  ShyLU_NodeTacho_Tacho_TestOpenMPDoubleChaseLevTaskScheduler_MPI_1 for intel-18 that we have seen fail before)

This is ready to merge!

</details>

**TASKS:**

* Reproduce problem locally [Done]

* Run full cee-rhel6-clang driver and install locally [Done]

* Update script docuemntation [Done]

* Create PR [Done]

* Manually merge to 'atdm-nightly-manual-updates' [Done]

**Comment by bartlettroscoe** [comment]
@sebrowne,

Looks like 'waterman' still has SuperLUDist 4.3.0:

```
/home/projects/ppc64le-pwr9-nvidia/superlu-dist/4.3.0/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/lib/libsuperlu_dist.a;
```

is that correct?  Is that going to get updated to SuperLUDist 5.4.0 as well?

**Comment by bartlettroscoe** [comment]
CC: @jfike, @sebrowne, @micahahoward,

'cee-rhel6' builds/installs are fixed and the PR #6048 has been merged to 'develop'.

Closing as complete.

----

As shown in automated testing today [here](https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2019-10-05&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-cee-rhel6) these builds and installs are fixed.

One can also see the full installs on the CEE LAN in:

```
$ du -sh /projects/atdm_devops/trilinos_installs/2019-10-05/*
1.2G    /projects/atdm_devops/trilinos_installs/2019-10-05/cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt
300M    /projects/atdm_devops/trilinos_installs/2019-10-05/cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt
1.9G    /projects/atdm_devops/trilinos_installs/2019-10-05/cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt
1.8G    /projects/atdm_devops/trilinos_installs/2019-10-05/cee-rhel6_intel-19.0.3_intelmpi-2018.4_serial_static_opt
```

NOTE: The `cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt` installed is smaller than the others because it uses shared libraries.  (Pretty massive difference)

The 


???

**Comment by sebrowne** [comment]
@bartlettroscoe,
No, that is not correct for waterman, but it should be fixed soon.

-Sam
